server/controllers/board.js

```diff
@@ -16,12 +18,24 @@ exports.fullBoardById = asyncHandler(async (req, res, next) => {
   }
 });
 
-// @route PUT /board/update
-// @desc Update board by reordering columns and/or cards
+// @route POST /board/update
+// @desc Update board with new columns and/or cards
 // @access Private
 exports.updateBoard = asyncHandler(async (req, res, next) => {
   try {
     const newBoard = req.body.board;
+    if (newBoard.columns) {
+      newBoard.columns.forEach(async column => {
+        if (column.cards) {
+          column.cards.forEach(async card => {
+            await Card.findByIdAndUpdate(card._id, card).exec();
+          });
+          column.cards = column.cards.map(card => card._id);
+        }
+        await Column.findByIdAndUpdate(column._id, column).exec();
+      });
+      newBoard.columns = newBoard.columns.map(column => column._id);
+    }
```

Comment:
This is an interesting way to do this however, not the greatest. Given timeline it's not critical but in future I think you could utilize more of the database driver more as opposed to the looping approach. Also, using async/await inside forEach is not the best thing to do.